src/common/MessageDispatcher.sol

```diff
@@ -281,11 +285,13 @@ contract MessageDispatcher is Auth, IMessageDispatcher {
         PoolId poolId,
         ShareClassId scId,
         bytes32 target,
-        bytes calldata payload
+        bytes calldata payload,
+        uint128 extraGasLimit
```

Comment:
Q: AFAICT, right now there is no need to have a non-zero extra gas limit for `UpdateContract` messages right? So this adjustment is just future safety or a I missing something?

I basically want to understand why it is added for the changed cases here. The `sendUpdateRestriction` makes totally sense to me but for the others I need some more input.